### PR TITLE
feat(workflows): describe workflow by name

### PR DESCRIPTION
**Pull request by jiparis**
This change in the API allows to describe workflows by name instead of (or in addition to) IDs. This enables use cases like discovering workflows based on a name (a github worflow name), useful in automated onboarding scenarios. 

`wf describe` command has now a new `--name` option to describe a workflow by name:
```
> go run app/cli/main.go --insecure wf describe --name mywf -o json
WRN API contacted in insecure mode
[
   {
      "name": "mywf",
      "id": "e6000b9f-9491-4a9f-9ab3-4988f5dee2ee",
      "team": "",
      "project": "myproj",
      "createdAt": "2024-04-23T17:32:42.529858Z",
      "runsCount": 0,
      "contractID": "46052e26-ea64-470a-a7ec-dd349b6178ad",
      "contractRevisionLatest": 1,
      "public": false
   }
]
```
```
> go run app/cli/main.go --insecure wf describe
WRN API contacted in insecure mode
ERR at least one of the flags in the group [id name] is required
exit status 1
```
The change is backwards compatible:
```
> go run app/cli/main.go --insecure wf describe --id e6000b9f-9491-4a9f-9ab3-4988f5dee2ee
WRN API contacted in insecure mode
┌──────────────────────────────────────┬──────┬─────────┬─────────────────────┬────────┬─────────────────┐
│ ID                                   │ NAME │ PROJECT │ CREATED AT          │ RUNNER │ LAST RUN STATUS │
├──────────────────────────────────────┼──────┼─────────┼─────────────────────┼────────┼─────────────────┤
│ e6000b9f-9491-4a9f-9ab3-4988f5dee2ee │ mywf │ myproj  │ 23 Apr 24 17:32 UTC │        │                 │
└──────────────────────────────────────┴──────┴─────────┴─────────────────────┴────────┴─────────────────┘
```
Refs #757

### CODE REVIEW

File: app/controlplane/api/controlplane/v1/workflow.proto [hunk 1]
```diff
@@ -72,7 +72,8 @@ message WorkflowServiceListResponse {
 }
 
 message WorkflowServiceViewRequest {
-  string id = 1 [(buf.validate.field).string.uuid = true];
+  optional string id = 1;
```
Comment (migmartri) [inline app/controlplane/api/controlplane/v1/workflow.proto:75]: You might want to add this validation. BTW why did you decide to make them optional? To me optional is useful when you want to differentiate between having an empty value or not-set value. In this case, it doesn't make a difference right?

https://github.com/chainloop-dev/chainloop/blob/4dac4f6150ccb9d42574c7d9e878401a414784ac/app/controlplane/api/controlplane/v1/workflow_contract.proto#L59

File: app/controlplane/internal/service/workflow.go [hunk 2]
```diff
@@ -123,7 +124,15 @@ func (s *WorkflowService) View(ctx context.Context, req *pb.WorkflowServiceViewR
...
 	}
 
-	wf, err := s.useCase.FindByIDInOrg(ctx, currentOrg.ID, req.Id)
+	var wf *biz.Workflow
+	if req.Name != nil {
+		wf, err = s.useCase.FindByNameInOrg(ctx, currentOrg.ID, *req.Name)
+	} else if req.Id != nil {
+		wf, err = s.useCase.FindByIDInOrg(ctx, currentOrg.ID, *req.Id)
+	} else {
+		return nil, errors.BadRequest("invalid", "either workflow ID or Name is required")
```
Comment (migmartri) [inline app/controlplane/internal/service/workflow.go:135]: as I mentioned we could add an API validation

**Review by migmartri — APPROVED** [review]
Mostly LGTM!

File: app/controlplane/api/controlplane/v1/workflow.proto (hunk 1 above)
Comment (jiparis) [reply app/controlplane/api/controlplane/v1/workflow.proto:75]: Checking that example from workflow contracts.

File: app/controlplane/internal/service/workflow.go (hunk 2 above)
Comment (jiparis) [reply app/controlplane/internal/service/workflow.go:135]: Done!

**Pull request merged**